string[] init_state
string[] goal_state
---
string[] solution